packages/connectivity/src/scp-cf/environment-accessor/xsuaa.ts

```diff
@@ -17,8 +17,10 @@ export function getXsuaaServiceCredentials(
     'xsuaa',
     token
   );
-  if (!credentials) {
-    throw new Error('Could not find binding to the XSUAA service.');
+  if (!credentials && token) {
+    throw new Error('Could not find a XSUAA service binding matching the token.');
+  } else if (!credentials) {
+    throw new Error('Could not find a XSUAA service binding.');
```

Comment:
[pp] This checks the same condition (`!credentials`) twice, I think this could be easier to read with the following:

```suggestion
  if (!credentials) {
    throw new Error(token ? 'Could not find a XSUAA service binding matching the token.' : 'Could not find a XSUAA service binding.');
```